### PR TITLE
throttling: fixup interactions with Timeline::get_vectored

### PR DESCRIPTION
**Pull request by problame**
## Problem

Before this PR, `Timeline::get_vectored` would be throttled twice if the sequential option was enabled or if validation was enabled.

Also, `pageserver_get_vectored_seconds` included the time spent in the throttle, which turns out to be undesirable for what we use that metric for.

## Summary of changes

Double-throttle:

* Add `Timeline::get0` method which is unthrottled.
* Use that method from within the `Timeline::get_vectored` code path.

Metric:

* return throttled time from `throttle()` method
* deduct the value from the observed time
* globally rate-limited logging of duration subtraction errors, like in all other places that do the throttled-time deduction from observations

### CODE REVIEW

**Comment by problame** [comment]
Release build fails reproducible with

```
error: queries overflow the depth limit!
  --> /__w/neon/neon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.36.0/src/runtime/task/raw.rs:72:5
   |
72 |     const TRAILER_OFFSET: usize = get_trailer_offset(
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: consider increasing the recursion limit by adding a `#![recursion_limit = "256"]` attribute to your crate (`pageserver`)
   = note: query depth increased by 130 when computing layout of `tokio::runtime::task::core::Core<{async fn body@pageserver::task_mgr::task_wrapper<{async block@pageserver::consumption_metrics::collect_metrics::{closure#0}::{closure#0}}>::{closure#0}}, alloc::sync::Arc<tokio::runtime::scheduler::multi_thread::handle::Handle>>`

error: could not compile `pageserver` (bin "pageserver") due to 1 previous error
warning: build failed, waiting for other jobs to finish...
Error: Process completed with exit code 101.
```

However, we already _have_ a `recursion_limit` on the `pageserver` crate:

https://github.com/neondatabase/neon/blob/cd449d66ea29ad2d7269458e90623c3ae40e1816/pageserver/src/lib.rs#L1

Bumping it to 600 doesn't help on my machine.

I don't know what to do.

**Comment by problame** [comment]
Got help from Arpad 🙏 https://neondb.slack.com/archives/C0277TKAJCA/p1710175354280539

File: pageserver/src/tenant/timeline.rs [hunk 1]
```diff
@@ -644,18 +646,27 @@ impl Timeline {
...
         &self,
         key: Key,
         lsn: Lsn,
         ctx: &RequestContext,
+    ) -> Result<Bytes, PageReconstructError> {
+        self.timeline_get_throttle.throttle(ctx, 1).await;
+        self.get0(key, lsn, ctx).await
+    }
+    /// Not subject to [`Self::timeline_get_throttle`].
+    async fn get0(
```
Comment (VladLazar) [inline pageserver/src/tenant/timeline.rs:660]: nit: I'd go for `get_impl` to have it simetrical with `get_vectored_impl`

**Review by VladLazar — APPROVED** [review]

File: pageserver/src/tenant/timeline.rs (hunk 1 above)
Comment (problame) [reply pageserver/src/tenant/timeline.rs]: 7e5c93fc70b6fd824a64b047aefbd6f8801c48b7

**Pull request merged**